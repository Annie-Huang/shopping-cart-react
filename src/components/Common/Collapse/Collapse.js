import React from 'react';
import PropTypes from 'prop-types';

import AnimateHeight from 'react-animate-height';

const Collapse = ({ children, expanded }) => (
    <AnimateHeight duration={400} height={expanded ? 'auto' : 0}>
        {children}
    </AnimateHeight>
);

Collapse.propTypes = {
    children: PropTypes.node.isRequired,
    expanded: PropTypes.bool.isRequired
};

export default Collapse;
