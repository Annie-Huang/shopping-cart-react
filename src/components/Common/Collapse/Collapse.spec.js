import React from 'react';
import { shallow } from 'enzyme';

import Collapse from './Collapse';

describe('<Collapse />', () => {
    it('renders expanded', () => {
        const expanded = true;
        const tree = shallow(<Collapse expanded={expanded}>Some content</Collapse>);
        expect(tree).toMatchSnapshot();
    });
    it('renders collapsed', () => {
        const expanded = false;
        const tree = shallow(<Collapse expanded={expanded}>some content</Collapse>);
        expect(tree).toMatchSnapshot();
    });
});
