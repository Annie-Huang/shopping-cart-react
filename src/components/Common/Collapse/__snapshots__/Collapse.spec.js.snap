// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapse /> renders collapsed 1`] = `
<AnimateHeight
  animateOpacity={false}
  animationStateClasses={
    Object {
      "animating": "rah-animating",
      "animatingDown": "rah-animating--down",
      "animatingToHeightAuto": "rah-animating--to-height-auto",
      "animatingToHeightSpecific": "rah-animating--to-height-specific",
      "animatingToHeightZero": "rah-animating--to-height-zero",
      "animatingUp": "rah-animating--up",
      "static": "rah-static",
      "staticHeightAuto": "rah-static--height-auto",
      "staticHeightSpecific": "rah-static--height-specific",
      "staticHeightZero": "rah-static--height-zero",
    }
  }
  applyInlineTransitions={true}
  delay={0}
  duration={400}
  easing="ease"
  height={0}
  style={Object {}}
>
  some content
</AnimateHeight>
`;

exports[`<Collapse /> renders expanded 1`] = `
<AnimateHeight
  animateOpacity={false}
  animationStateClasses={
    Object {
      "animating": "rah-animating",
      "animatingDown": "rah-animating--down",
      "animatingToHeightAuto": "rah-animating--to-height-auto",
      "animatingToHeightSpecific": "rah-animating--to-height-specific",
      "animatingToHeightZero": "rah-animating--to-height-zero",
      "animatingUp": "rah-animating--up",
      "static": "rah-static",
      "staticHeightAuto": "rah-static--height-auto",
      "staticHeightSpecific": "rah-static--height-specific",
      "staticHeightZero": "rah-static--height-zero",
    }
  }
  applyInlineTransitions={true}
  delay={0}
  duration={400}
  easing="ease"
  height="auto"
  style={Object {}}
>
  Some content
</AnimateHeight>
`;
